lib/init.lua

```diff
@@ -96,13 +97,14 @@ Promise.Status = {
 			end)
 ]]
 function Promise.new(callback)
-	local promise = {
+	local self = setmetatable({
 		-- Used to locate where a promise was created
 		_source = debug.traceback(),
 
 		-- A tag to identify us as a promise
 		_type = "Promise",
 
+		-- The status of the Promise
```

Comment:
This comment doesn't add anything to the code. It's a field called status on a promise, I think we can safely assume it's the status of the promise!